node/actors/network/src/gossip/batch_votes.rs

```diff
@@ -125,10 +130,12 @@ impl BatchVotes {
             }
 
             // Check the signature before insertion.
-            d.verify()?;
-
-            self.add(d.clone(), weight);
-            stats.added(d.msg.number, weight);
+            if let Err(e) = d.verify() {
+                tracing::error!(error =? e, "failed to verify batch vote");
+            } else {
```

Comment:
nit: I guess this operator logs error with "{:?}" formatting, which afair just shows the outermost message in case of anyhow::Errors (if it indeed is an anyhow error), "{:#}" is more bullet proof, so I recommend using it everywhere.

Reply:
Yeah, it just doesn't work with KV pairs, or I don't know how to use it. I'll just log it as part of the message.